File: 0202-happy-number/0202-happy-number.java
class Solution 
{
    public boolean isHappy(int n) 
    {
        int x = n;
        while (x != 1 && x != 4)
        {
            int S = 0;
            for (int i = x; i > 0; i /= 10)
            {
                int j = i % 10;
                S += (int)Math.pow(j, 2);
            }
            x = S;
        }
        return (x == 1);
    }
}

File: 0202-happy-number/0202-happy-number.py
class Solution:
    def isHappy(self, n):
        x = n
        while x != 1 and x != 4:
            S = 0
            i = x
            while i > 0:
                j = i % 10
                S += j ** 2
                i //= 10
            x = S
        return x == 1